From 1cb5fa2c1968e5951f9898ffd99a6255dd726c42 Mon Sep 17 00:00:00 2001
From: Levi Zim <rsworktech@outlook.com>
Date: Sun, 15 Jun 2025 22:37:31 -0700
Subject: [PATCH] [Updater] Use ProcessCPUArchitecture

The BuildArch() function added in updater.cc only supports arm64, x86
and x86_64, while breaking the build for all other architectures.

Use base::SysInfo::ProcessCPUArchitecture instead of inventing a new one
to fix it.

Change-Id: I7bc118b25333ece889b0714c0f321fc983220215
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6634009
Reviewed-by: Noah Rose Ledesma <noahrose@google.com>
Auto-Submit: Levi Zim <rsworktech@outlook.com>
Commit-Queue: Joshua Pawlicki <waffles@chromium.org>
Reviewed-by: Lei Zhang <thestig@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1474160}
---
 chrome/updater/updater.cc | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/chrome/updater/updater.cc b/chrome/updater/updater.cc
index f7dca12190d56..4a67813b6c48e 100644
--- a/chrome/updater/updater.cc
+++ b/chrome/updater/updater.cc
@@ -262,18 +262,6 @@ constexpr const char* BuildFlavor() {
 #endif
 }
 
-constexpr const char* BuildArch() {
-#if defined(ARCH_CPU_ARM64)
-  return "64 bit (ARM)";
-#elif defined(ARCH_CPU_X86_64)
-  return "64 bit (x64)";
-#elif defined(ARCH_CPU_X86)
-  return "32 bit (x86)";
-#else
-#error CPU architecture is unknown.
-#endif
-}
-
 std::string OperatingSystemVersion() {
 #if BUILDFLAG(IS_WIN)
   const base::win::OSInfo::VersionNumber v =
@@ -326,7 +314,8 @@ int UpdaterMain(int argc, const char* const* argv) {
   const UpdaterScope updater_scope = GetUpdaterScope();
   InitLogging(updater_scope);
   VLOG(1) << "Version: " << kUpdaterVersion << ", " << BuildFlavor() << ", "
-          << BuildArch() << ", command line: " << GetCommandLineString();
+          << base::SysInfo::ProcessCPUArchitecture()
+          << ", command line: " << GetCommandLineString();
   VLOG(1) << "OS version: " << OperatingSystemVersion()
           << ", System uptime (seconds): "
           << base::SysInfo::Uptime().InSeconds() << ", parent pid: "
-- 
2.39.5

